bundle/lib/source/DobbySpecConfig.cpp

```diff
@@ -2537,6 +2548,115 @@ bool DobbySpecConfig::processCapabilities(const Json::Value& value,
     return true;
 }
 
+// -----------------------------------------------------------------------------
+/**
+ *  @brief Processes the seccomp field of the json spec
+ *
+ *  Example json:
+ *
+ *      "seccomp": {
+ *          "defaultAction": "SCMP_ACT_ALLOW",
+ *          "syscalls": {
+ *              "names": [
+ *                  "getcwd",
+ *                  "chmod"
+ *              ],
+ *              "action": "SCMP_ACT_ERRNO"
+ *          }
+ *      }
+ *
+ *  This adds extra capabilities to the container.
+ *
+ *  @param[in]  value       The json spec document from the client
+ *  @param[in]  dictionary  Pointer to the OCI dictionary to populate
+ *
+ *  @return true if correctly processed the value, otherwise false.
+ */
+bool DobbySpecConfig::processSeccomp(const Json::Value& value,
+                                    ctemplate::TemplateDictionary* dictionary)
+{
+    if (!value.isObject())
+    {
+        AI_LOG_ERROR("invalid 'seccomp' field");
+        return false;
+    }
+
+    const Json::Value& defaultAction = value["defaultAction"];
+    if (!processSeccompAction(defaultAction))
+    {
+        AI_LOG_ERROR("invalid 'seccomp.defaultAction' field");
+        return false;
+    }
+
+    const Json::Value& syscalls = value["syscalls"];
+    if (!syscalls.isObject())
+    {
+        AI_LOG_ERROR("invalid 'seccomp.syscalls' field");
+        return false;
+    }
+
+    const Json::Value& action = syscalls["action"];
+    if (!processSeccompAction(action))
+    {
+        AI_LOG_ERROR("invalid 'seccomp.syscalls.action' field");
+        return false;
+    }
+
+    const Json::Value& names = syscalls["names"];
+    if (!names.isArray())
+    {
+        AI_LOG_ERROR("invalid 'seccomp.syscalls.names' field");
+        return false;
+    }
+
+    std::stringstream ss;
+    if (names.size() > 0)
+    {
+        for (int i = 0; i < (int)names.size() - 1; ++i)
+        {
+            const Json::Value& entry = names[i];
+            if (!entry.isString())
+            {
+                AI_LOG_ERROR("invalid 'seccomp.syscalls.names[%d]' field", i);
+                return false;
+            }
+
+            ss << "\"" << entry.asString() << "\", ";
+        }
+
+        ss << "\"" << names[names.size() - 1].asString() << "\"";
+    }
+    else
+    {
+        AI_LOG_ERROR("empty 'seccomp.syscalls.names' array");
+        return false;
+    }
+
+    dictionary->SetValue(SECCOMP_DEFAULT_ACTION, defaultAction.asString());
+    dictionary->SetValue(SECCOMP_ACTION, action.asString());
+    dictionary->SetValue(SECCOMP_SYSCALLS, ss.str());
+    dictionary->ShowSection(SECCOMP_ENABLED);
+
+    return true;
+}
+
+bool DobbySpecConfig::processSeccompAction(const Json::Value& value) const
```

Comment:
what is the purpose of this function ?  can you add a brief description and doxygen format comment ?

Reply:
Comment added. Also renamed the method to validateSeccompAction.